src/main/java/ch/njol/skript/expressions/ExprBlockSound.java

```diff
@@ -0,0 +1,125 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.Sound;
+import org.bukkit.SoundGroup;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+@Name("Block Sound")
+@Description("Gets the sound that a given block, blockdata, or itemtype will use in a specific scenario.")
+@Examples({
+	"play sound (break sound of dirt) at all players",
+	"set {_sounds::*} to place sounds of dirt, grass block, blue wool and stone"
+})
+@Since("INSERT VERSION")
+public class ExprBlockSound extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprBlockSound.class, String.class, ExpressionType.COMBINED,
+			"[the] break sound[s] of %blocks/blockdatas/itemtypes%",
+			"%blocks/blockdatas/itemtypes%'[s] break sound[s]",
+
+			"[the] fall sound[s] of %blocks/blockdatas/itemtypes%",
+			"%blocks/blockdatas/itemtypes%'[s] fall sound[s]",
+
+			"[the] hit sound[s] of %blocks/blockdatas/itemtypes%",
+			"%blocks/blockdatas/itemtypes%'[s] hit sound[s]",
+
+			"[the] place sound[s] of %blocks/blockdatas/itemtypes%",
+			"%blocks/blockdatas/itemtypes%'[s] place sound[s]",
+
+			"[the] step sound[s] of %blocks/blockdatas/itemtypes%",
+			"%blocks/blockdatas/itemtypes%'[s] step sound[s]");
```

Comment:
Would it not be easier to combine the patterns and use a `1: 2: 3:...` parse mark?